sourcegraph/learn
# Updates to ToC header highlighting

### What should this PR do?
Resolves [DEVED-184](https://sourcegraph.atlassian.net/browse/DEVED-184) by adding highlighting to ToC header links on click, in addition to the existing highlighting on scroll behavior.

Also updates the top-level element in `HeaderLink` to an `a` rather than a Next.js `Link`, which was throwing some Swiftype warnings in the console.

### Why are we making this change?
- We want users to be able to track their current place in tutorials as they read.

### What are the acceptance criteria? 
- Existing scrolling behavior should work as expected on scroll.
- Clicking on headers should highlight them.

### How should this PR be tested?
- Check out the branch and test that scrolling highlighting works properly in both directions.
- Click on headers and check that they are highlighted.
- Check console to ensure that there are no Swiftype warnings on header clicks.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
